src/cleed++/IVCurve.cc

```
@@ -0,0 +1,280 @@
+IVCurve& IVCurve::setIVData(const vector<real> &x,
+                            const vector<real> &y,
+                            const vector<real> &deriv2) {
+  std::size_t n = (x.size() < y.size()) ? x.size() : y.size();
+
+  rfac_iv_data *data = new rfac_iv_data[n];
+  for (std::size_t i=0; i < n; i++ ) {
+    data[i].energy = x[i];
+    data[i].intens = y[i];
+  }
+
+  if (n < deriv2.size()) {
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)